graylog2-server/src/main/java/org/graylog2/indexer/cluster/Cluster.java

```diff
@@ -90,64 +112,83 @@ public ClusterHealthResponse health() {
+     * @return A {@link JsonArray} with the result of the cat nodes request.
+     */
+    private JsonArray catNodes(String... fields) {
+        final String fieldNames = Strings.join(fields, ",");
+        final Cat request = new Cat.NodesBuilder()
+                .setParameter("h", fieldNames)
+                .setParameter("full_id", true)
+                .build();
+        final CatResult response;
+        try {
+            response = jestClient.execute(request);
+        } catch (IOException e) {
+            throw new ElasticsearchException("Unable to read Elasticsearch node information", e);
         }
 
-        return builder.build();
+        if (response.isSucceeded()) {
+            return GsonUtils.asJsonArray(response.getJsonObject().get("result"));
+        } else {
+            throw new ElasticsearchException("Unable to read Elasticsearch node information");
+        }
     }
 
-    public Map<String, NodeStats> getNodesStats(String... nodesIds) {
-        final ClusterAdminClient clusterAdminClient = c.admin().cluster();
-        final NodesStatsRequest request = clusterAdminClient.prepareNodesStats(nodesIds).request();
-        final ImmutableMap.Builder<String, NodeStats> builder = ImmutableMap.builder();
-        for (NodeStats nodeStats : clusterAdminClient.nodesStats(request).actionGet().getNodes()) {
-            builder.put(nodeStats.getNode().id(), nodeStats);
+    public Set<NodeFileDescriptorStats> getFileDescriptorStats() {
+        final JsonArray nodes = catNodes("name", "host", "fileDescriptorMax");
+        final ImmutableSet.Builder<NodeFileDescriptorStats> setBuilder = ImmutableSet.builder();
+        for (JsonElement jsonElement : nodes) {
+            if (jsonElement.isJsonObject()) {
+                final JsonObject jsonObject = jsonElement.getAsJsonObject();
+                final String name = GsonUtils.asString(jsonObject.get("name"));
+                final String host = GsonUtils.asString(jsonObject.get("host"));
+                final Long maxFileDescriptors = GsonUtils.asLong(jsonObject.get("fileDescriptorMax"));
+                setBuilder.add(NodeFileDescriptorStats.create(name, host, maxFileDescriptors));
+            }
         }
 
-        return builder.build();
+        return setBuilder.build();
     }
 
     public String nodeIdToName(String nodeId) {
-        final NodeInfo nodeInfo = getNodeInfo(nodeId);
-        return nodeInfo == null ? "UNKNOWN" : nodeInfo.getNode().getName();
-
+        return getNodeInfo(nodeId)
+                .map(nodeInfo -> GsonUtils.asString(nodeInfo.get("name")))
+                .orElse("UNKNOWN");
     }
 
     public String nodeIdToHostName(String nodeId) {
-        final NodeInfo nodeInfo = getNodeInfo(nodeId);
-        return nodeInfo == null ? "UNKNOWN" : nodeInfo.getHostname();
+        return getNodeInfo(nodeId)
+                .map(nodeInfo -> GsonUtils.asString(nodeInfo.get("host")))
+                .orElse("UNKNOWN");
```

Comment:
See above, plus: `"UNKNOWN"` is a valid host name. If someone ever wants to actually use the result of this beyond displaying it, that could be rather nasty.